.github/workflows/ci.yaml

```diff
@@ -21,14 +21,14 @@ jobs:
       run: |
         cp cabal.project.werror cabal.project.local
         cp cabal.project.freeze.ghc-${{ matrix.ghc-ver }} cabal.project.freeze
-    - uses: haskell/actions/setup@v1
+    - uses: haskell-actions/setup@v2
       id: setup-haskell
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc-ver }}
         cabal-version: ${{ matrix.cabal }}
     - name: Cache
-      uses: actions/cache@v1
+      uses: actions/cache@v4
```

Comment:
Worth upgrading the version of `actions/checkout` (used above, which GitHub won't let me add a review comment for) to `@v4` while you're in the neighborhood.